homeassistant/loader.py

```diff
@@ -142,6 +142,63 @@ async def async_get_config_flows(hass: "HomeAssistant") -> Set[str]:
     return flows
 
 
+async def async_get_zeroconf(hass: "HomeAssistant") -> Dict[str, List]:
+    """Return cached list of zeroconf types."""
+    # pylint: disable=import-outside-toplevel
+    from homeassistant.generated.zeroconf import ZEROCONF
+
+    zeroconf: Dict[str, List] = ZEROCONF.copy()
+
+    integrations = await async_get_custom_components(hass)
+    for integration in integrations.values():
+        if not integration.zeroconf:
+            continue
+        for typ in integration.zeroconf:
+            zeroconf.setdefault(typ, [])
+            if integration.domain not in zeroconf[typ]:
+                zeroconf[typ].append(integration.domain)
+
+    return zeroconf
+
+
+async def async_get_homekit(hass: "HomeAssistant") -> Dict[str, str]:
+    """Return cached list of homekit models."""
+    # pylint: disable=import-outside-toplevel
+    from homeassistant.generated.zeroconf import HOMEKIT
+
+    homekit: Dict[str, str] = HOMEKIT.copy()
+
+    integrations = await async_get_custom_components(hass)
+    for integration in integrations.values():
+        if (
+            not integration.homekit
+            or "models" not in integration.homekit
+            or not integration.homekit["models"]
+        ):
+            continue
+        for model in integration.homekit["models"]:
+            homekit[model] = integration.domain
+
+    return homekit
+
+
+async def async_get_ssdp(hass: "HomeAssistant") -> Dict[str, List]:
```

Comment:
@ludeeus in a future PR, you might want to upgrade the discoveries to add a cache that HACS can clear after it installs a new integration 👍

Reply:
We can probably add a hook to restart the zeroconf service browser as well.